xarray/core/variable.py

```diff
@@ -2096,12 +2098,12 @@ def concat(
         encoding = dict(first_var.encoding)
         if not shortcut:
             for var in variables:
-                if var.dims != first_var.dims:
+                if var.dims != first_var_dims:
                     raise ValueError(
-                        f"Variable has dimensions {list(var.dims)} but first Variable has dimensions {list(first_var.dims)}"
+                        f"Variable has dimensions {list(var.dims)} but first Variable has dimensions {list(first_var_dims)}"
                     )
 
-        return cls(dims, data, attrs, encoding)
+        return cls(dims, data, attrs, encoding, fastpath=True)
```

Comment:
So the fastpath is OK here because know `data` is a duck array? I'm somewhat confused by the fastpath.

Reply:
Yeah, my idea is that since `Variable` is an input to the method we can assume `data` is valid. 

I think we can do this in most places where xarray recreates a `Variable`.